kvbc/src/Replica.cpp

```diff
@@ -448,6 +451,41 @@ bool Replica::getBlock(uint64_t blockId, char *outBlock, uint32_t *outBlockSize)
   return true;
 }
 
+std::future<bool> Replica::getBlockAsync(uint64_t blockId, char *outBlock, uint32_t *outBlockSize) {
+  static uint64_t callCounter = 0;
+  static constexpr size_t snapshotThresh = 1000;
+
+  auto future = blocksIOWorkersPool_.async(
+      [this](uint64_t blockId, char *outBlock, uint32_t *outBlockSize) {
+        auto start = std::chrono::steady_clock::now();
+        *outBlockSize = 0;
+
+        // getBlock will throw exception if block doesn't exist
+        try {
+          getBlock(blockId, outBlock, outBlockSize);
+        } catch (NotFoundException &ex) {
+          *outBlockSize = 0;
```

Comment:
LOG_ERROR

Reply:
Will ads an error, note: I have some errors outside as well in the source (when the return value is false).